pkg/apic/provisioning/requeststatus.go

```diff
@@ -0,0 +1,69 @@
+package provisioning
+
+import "fmt"
+
+const statusSetError = "can not set status as it's already been set"
+
+// RequestStatus - holds info about the status of the request
+type RequestStatus struct {
+	status  status
+	message string
+}
+
+// RequestStatusBuilder - builder to create new request status
+type RequestStatusBuilder interface {
+	Success() RequestStatusBuilder
+	Failed(message string) RequestStatusBuilder
+	Process() (*RequestStatus, error)
+}
+
+type requestStatusBuilder struct {
+	err    error
+	status *RequestStatus
+}
+
+// NewRequestStatusBuilder - create a request status builder
+func NewRequestStatusBuilder() RequestStatusBuilder {
+	return &requestStatusBuilder{
+		status: &RequestStatus{},
+	}
+}
+
+// Process - process the builder, returning errors
+func (r *requestStatusBuilder) Process() (*RequestStatus, error) {
```

Comment:
Do we need this, we could directly return the status from Success or Failed methods. Also not sure of we need to report error. Trying to see how the agents will call it and it does not seem right

```
func (a *Subscription) ApplicationRequestDeprovision(applicationRequest provisioning.ApplicationRequest) (status provisioning.RequestStatus) {
    ...
    status, err := builder.Success().Process()
    if err != nil {
        // what should the agent do ?
    }
    return status
}
```

Would it be much easier for the agent to call 
```
func (a *Subscription) ApplicationRequestDeprovision(applicationRequest provisioning.ApplicationRequest) (status provisioning.RequestStatus) {
    ...
    return builder.Success()
}
```

Reply:
Changed it, removed process method. Return the RequestStatus interface directly